src/codemagic/tools/firebase_app_distribution/firebase.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+import argparse
+from typing import Dict
+
+from codemagic import cli
+from codemagic.firebase.client import FirebaseClient
+
+from .action_groups import ReleasesActionGroup
+from .actions import GetLatestBuildVersionAction
+from .arguments import FirebaseArgument
+
+
+@cli.common_arguments(FirebaseArgument.PROJECT_ID, FirebaseArgument.FIREBASE_SERVICE_ACCOUNT_CREDENTIALS)
+class FirebaseAppDistribution(
+    cli.CliApp,
+    GetLatestBuildVersionAction,
+    ReleasesActionGroup,
+):
+    """
+    Utility to list releases and retrieve the latest release build version number from Firebase
+    """
+
+    def __init__(self, credentials: Dict, project_id: str, **kwargs):
+        super().__init__(**kwargs)
+        self.client = FirebaseClient(credentials)
+        self._project_id = project_id
+
+    @property
+    def project_id(self):
+        return self._project_id
```

Comment:
What is the benefit on this over just
```suggestion
        self.project_id = project_id
```